packages/framework/src/Form/Constraints/MoneyRange.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Form\Constraints;
+
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @Annotation
+ */
+class MoneyRange extends Constraint
+{
+    /**
+     * @var string
+     */
+    public $minMessage = 'The amount of money should be {{ limit }} or more.';
+
+    /**
+     * @var string
+     */
+    public $maxMessage = 'The amount of money should be {{ limit }} or less.';
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Money\Money|null
+     */
+    public $min;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Money\Money|null
+     */
+    public $max;
+
+    /**
+     * @param array|null $options
+     */
+    public function __construct($options = null)
+    {
+        parent::__construct($options);
+
+        if ($this->min === null && $this->max === null) {
+            $message = sprintf('Either option "min" or "max" must be given for constraint "%s".', __CLASS__);
+
+            throw new \Symfony\Component\Validator\Exception\MissingOptionsException($message, ['min', 'max']);
+        }
+
+        $this->validateMoneyOrNullOption('min', $this->min);
```

Comment:
This is strange in combination with typehint `mixed $value` in `validateMoneyOrNullOption` because we are passing `$this->min` that is `?Money`

I would prefer to validate options before calling parent constructor - local variable will be passed and `$this->min/max` will be always `?Money`

Reply:
I get your point, rewritten in 893142f1aad654362e0ef7d8edffacb84acfa544